### ISSUE TITLE
Getting stuck in Created CPE Index 

### ISSUE THREAD

**Issue by drizzling4u** [dependency-check/DependencyCheck#2745]
**Describe the bug**

**[INFO] Created CPE Index (938 seconds)
[INFO] Finished CPE Analyzer (938 seconds)**
[dependency_check.log](https://github.com/jeremylong/DependencyCheck/files/5030434/dependency_check.log)

**Version of dependency-check used**
The problem occurs using version 5.3.2  of the jenkins plugin  (cli, gradle plugin, maven plugin, etc.)

**Log file**
When reporting errors, 99% of the time log file output is required. Please post the log file as a [gist](https://gist.github.com/) and provide a link in the new issue.

**To Reproduce**
Using remote connectivity to oracle database.
below are the command used : local details are intentionally marked as "*******"

<
dependency-check/bin/dependency-check.sh --log /home/jenkins/org.jenkinsci.plugins.DependencyCheck.tools.DependencyCheckInstallation/dependency-check/project_latest.log --proxyserver ******************  --proxyport **  --project TEST --disableCentral --out "*******" --scan "*******"  --dbDriverName oracle.jdbc.OracleDriver --dbPassword "*******" --dbUser "*******" --dbDriverPath "*******"/ojdbc7.jar --connectionString jdbc:oracle:thin:/@"*******":1521:DEMODB --format XML
>

**Expected behavior**

**Additional context**

**Comment by drizzling4u** [comment]
Jenkins , scan was a success but took a lot of time while same was not taking from command line.

Picked up JAVA_TOOL_OPTIONS: -Xmx800M
[INFO] Checking for updates
[INFO] Skipping NVD check since last check was within 4 hours.
[INFO] Check for updates complete (2200 ms)
[INFO] 

Dependency-Check is an open source tool performing a best effort analysis of 3rd party dependencies; false positives and false negatives may exist in the analysis performed by the tool. Use of the tool and the reporting provided constitutes acceptance for use in an AS IS condition, and there are NO warranties, implied or otherwise, with regard to the analysis or its use. Any use of the tool and the reporting provided is at the user’s risk. In no event shall the copyright holder or OWASP be held liable for any damages whatsoever arising out of or in connection with the use of this tool, the analysis performed, or the resulting report.


[INFO] Analysis Started
[INFO] Finished Archive Analyzer (0 seconds)
[INFO] Finished File Name Analyzer (0 seconds)
[INFO] Finished Jar Analyzer (0 seconds)
[INFO] Finished Dependency Merging Analyzer (0 seconds)
[INFO] Finished Version Filter Analyzer (0 seconds)
[INFO] Finished Hint Analyzer (0 seconds)
_**[INFO] Created CPE Index (938 seconds)
[INFO] Finished CPE Analyzer (938 seconds)**_
[INFO] Finished False Positive Analyzer (0 seconds)
[INFO] Finished NVD CVE Analyzer (0 seconds)
[INFO] Finished RetireJS Analyzer (0 seconds)
[INFO] Finished Sonatype OSS Index Analyzer (2 seconds)
[INFO] Finished Vulnerability Suppression Analyzer (0 seconds)
[INFO] Finished Dependency Bundling Analyzer (0 seconds)
[INFO] Analysis Complete (941 seconds)
[Pipeline] }
[Pipeline] // withEnv
[Pipeline] }
[Pipeline] // stage
[Pipeline] }
[Pipeline] // withEnv
[Pipeline] }
[Pipeline] // node
[Pipeline] }
[Pipeline] // podTemplate
[Pipeline] End of Pipeline
Finished: SUCCESS

**Comment by jeremylong** [comment]
In this case most of the time was spent creating the CPE Index. When using an external database there are many factors in play - connection speed and distance between the scanning server and jenkins.

**Comment by drizzling4u** [comment]
I have tried 2 things. DB and jenkins are in both same server.
1) Tried using default h2db
This gets completed in 10 min including updated and creating indexes everything.
2) Tried using oracle DB
Update itself where using oracle is running from last 4 hours.

**Comment by drizzling4u** [comment]
I somehow understood the pattern, in case of oracle , i think we are making one inserts per network call and there is commit only once the complete set is inserted. 

![image](https://user-images.githubusercontent.com/54618199/89534190-2d586480-d812-11ea-88f5-ba23f615a019.png)


SQL_ID 5mt2pp3d59xk7
--------------------
SELECT id FROM vulnerability WHERE cve = :1
 
Plan hash value: 162675211
 
---------------------------------------------------------------------------------------------
| Id  | Operation                   | Name          | Rows  | Bytes | Cost (%CPU)| Time     |
---------------------------------------------------------------------------------------------
|   0 | SELECT STATEMENT            |               |       |       |     1 (100)|          |
|   1 |  TABLE ACCESS BY INDEX ROWID| VULNERABILITY |     1 |    25 |     1   (0)| 00:00:01 |
|   2 |   INDEX UNIQUE SCAN         | SYS_C007430   |     1 |       |     1   (0)| 00:00:01 |
---------------------------------------------------------------------------------------------
 
![image](https://user-images.githubusercontent.com/54618199/89534388-73152d00-d812-11ea-9557-a62d99c54393.png)

**Comment by drizzling4u** [comment]
executions are two high , i have just shared one select query ... There are combination of inserts/select/delete for each row. and while currently it is stuck in 

INFO  - Download Started for NVD CVE - 2020
2020-08-06 18:15:48,523 org.owasp.dependencycheck.utils.HttpResourceConnection:175
DEBUG - Attempting retrieval of https://nvd.nist.gov/feeds/json/cve/1.1/nvdcve-1.1-2020.json.gz
2020-08-06 18:15:48,567 org.owasp.dependencycheck.data.update.nvd.DownloadTask:139
INFO  - Download Complete for NVD CVE - 2019  (4142 ms)
2020-08-06 18:15:54,395 org.owasp.dependencycheck.data.update.nvd.DownloadTask:139
INFO  - Download Complete for NVD CVE - 2020  (5872 ms)

**Comment by drizzling4u** [comment]
@jeremylong if you can help me in getting the source and issue pointer, i can fix this. This seems to be issue while handling DB calls for update/delete/selects

**Comment by jeremylong** [comment]
With changes in flight for 6.0.0 the update time should get a lot better. However, we could use some help with the updates for Oracle - a few simple stored procedures need to be written. See #2755. These have already been written for H2 and MySQL - we still need Oracle, Postgresql, and Ms SQL updates completed.

**Comment by jeremylong** [comment]
I've completed the updates for the base H2 and MySQL and Postgresql.  Updates times I recorded creating a new database:

Database | Time in minutes
-- | --
h2 | 2.18948333
postgresql | 10.0877167
mysql | 23.8273333

However, we still need the updated table structure and stored procs implemented for Oracle. See #2755 and #2754.
